modules/gdscript/gdscript.cpp

```diff
@@ -152,12 +152,13 @@ Variant GDScript::_new(const Variant **p_args, int p_argcount, Variant::CallErro
 	}
 
 	ERR_FAIL_COND_V(_baseptr->native.is_null(), Variant());
-
-	if (_baseptr->native.ptr()) {
+	if (_baseptr->native->instance()) {
 		owner = _baseptr->native->instance();
 	} else {
-		owner = memnew(Reference); //by default, no base means use reference
+		owner = memnew(Reference);
 	}
+	ERR_EXPLAIN("Can't inherit from a virtual class");
+	ERR_FAIL_COND_V(!owner, Variant());
```

Comment:
Can this condition be true at all?
If `_baseptr->native->instance()` is `NULL`, then `owner` is `memnew(Reference)`, which shouldn't give `NULL`.
If `_baseptr->native->instance()` is not `NULL`, then `owner` is `_baseptr->native->instance()`.

Reply:
Ah yes, I put the wrong condition in the if. Will correct tomorrow-ish